### PR TITLE
feat(open-klant): digitale adressen partij fallback bij afwezigheid eigen adressen

### LINKED ISSUE #445: 🛠 Task: Voeg GetPartijDigitaleAdressenAsync toe aan OpenKlantApiClient
# 🛠 Task: Voeg `GetPartijDigitaleAdressenAsync` toe aan `OpenKlantApiClient`

---

## Intent

De medewerker moet de digitale adressen van een partij kunnen zien wanneer de betrokkene geen eigen adressen heeft. Hiervoor is een methode nodig die de digitale adressen van een partij ophaalt via de Open Klant API.

---

## Functional Description

Voeg een methode `GetPartijDigitaleAdressenAsync(string partijUuid)` toe aan `IOpenKlantApiClient` en de implementatie in `OpenKlantApiClient`. De methode roept `GET /partijen/{uuid}?expand=digitaleAdressen` aan en mapt het response naar een `List<DigitaleAdres>`. Breid het bestaande `Partij`-model uit met een `DigitaleAdressen` property.

---

## Design Reference

N/A — geen design voor deze taak.

---

### In scope

- Nieuwe methode `GetPartijDigitaleAdressenAsync` op `IOpenKlantApiClient`
- Implementatie in `OpenKlantApiClient` met `GET /partijen/{uuid}?expand=digitaleAdressen`
- Uitbreiden `Partij`-model met `List<DigitaleAdres>? DigitaleAdressen` property
- Logging conform bestaande patronen

### Out of scope

- Fallback-logica in `InternetaakDetailsService` (Task 2)
- Frontend wijzigingen
- Foutafhandeling bij onbereikbare API (Task 2 bepaalt strategie)

---

### Request

```
GET /partijen/{uuid}?expand=digitaleAdressen
```

### Response

```json
{
  "uuid": "string",
  "url": "string",
  "naam": "string",
  "_expand": {
    "digitaleAdressen": [
      {
        "uuid": "string",
        "url": "string",
        "adres": "string",
        "soortDigitaalAdres": "email|telefoonnummer",
        "isStandaardAdres": true,
        "omschrijving": "string"
      }
    ]
  }
}
```

### Events / Side Effects

Geen — read-only operatie.

---

## Technical Approach

- Voeg `DigitaleAdressen` property toe aan het bestaande `Partij`-model in `Klantcontact.cs`
- Voeg interface-methode toe aan `IOpenKlantApiClient`
- Implementeer in `OpenKlantApiClient` met `GetFromJsonAsync` + expand response mapping
- Gebruik bestaand `DigitaleAdres`-model (reeds gedefinieerd in `Klantcontact.cs:64`)

### Aggregates & Invariants

- Aggregates: Partij (read-only)
- Invariants: Geen — dit is een read operatie

---

## Impacted Areas

- Backend: `InterneTaakAfhandeling.Common/Services/OpenKlantApi/OpenKlantApiClient.cs`, `Models/Klantcontact.cs`
- Frontend: Geen
- Database: N/A
- APIs: Open Klant API (`/partijen/{uuid}?expand=digitaleAdressen`)
- External systems: Open Klant

---

## Gherkin

```gherkin
Feature: Digitale adressen ophalen van partij

  Scenario: Partij met digitale adressen
    Given een partij met een gekoppeld e-mailadres en telefoonnummer
    When de digitale adressen van de partij worden opgehaald
    Then worden het e-mailadres en telefoonnummer van de partij teruggegeven

  Scenario: Partij zonder digitale adressen
    Given een partij zonder gekoppelde digitale adressen
    When de digitale adressen van de partij worden opgehaald
    Then wordt een lege lijst teruggegeven

  Scenario: Partij niet gevonden
    Given een partij-identifier die niet bestaat in Open Klant
    When de digitale adressen van de partij worden opgehaald
    Then wordt een fout gelogd en een lege lijst teruggegeven
```

---

## Edge Cases & Risks

- Partij UUID verwijst naar een niet-bestaande partij → graceful degradation (log + return empty)
- Open Klant API geeft een expand-response zonder `_expand.digitaleAdressen` veld → behandel als lege lijst

---

## Observability

- Logs: `LogInformation` bij ophalen partij, `LogWarning` bij niet-gevonden of API-fout
- Metrics: Geen nieuwe metrics vereist
- Alerts: Geen

---

## Dependencies

Geen — dit is de eerste taak in de keten.

---

## Rollout

- Feature flag: N/A
- Backward compatibility: Toevoeging van optioneel veld aan model, bestaande code niet geraakt
- Data migration: N/A

---

## Definition of Done

- [ ] Contracts match implementation
- [ ] All Gherkin scenarios covered by tests (Phase 2 E2E)
- [ ] Edge cases handled
- [ ] Observability in place
- [ ] Code reviewed
- [ ] Linting passes (`npm run lint:ci`)
- [ ] Formatting passes (`npm run format:ci`)

---

## Test Mapping

| Gherkin Scenario | Test file | Status |
| ---------------- | --------- | ------ |
| Partij met digitale adressen | Phase 2 E2E | Pending |
| Partij zonder digitale adressen | Phase 2 E2E | Pending |
| Partij niet gevonden | Phase 2 E2E | Pending |

### LINKED ISSUE #446: 🛠 Task: Implementeer server-side fallback-logica digitale adressen in InternetaakDetailsService
# 🛠 Task: Implementeer server-side fallback-logica in InternetaakDetailsService

---

## Intent

De medewerker ziet altijd contactgegevens als deze beschikbaar zijn — ook wanneer ze niet bij de betrokkene zelf maar bij de gekoppelde partij zijn opgeslagen (OIP-scenario).

---

## Functional Description

Implementeer fallback-verrijking in `InternetaakDetailsService`: na het laden van het klantcontact met betrokkenen, controleer voor elke betrokkene of `_expand.digitaleAdressen` leeg is. Zo ja, en `_expand.wasPartij` een UUID bevat, haal digitale adressen op via `GetPartijDigitaleAdressenAsync` en injecteer deze in de betrokkene-expand. Het client-contract (digitaleAdressen array) blijft ongewijzigd.

---

## Design Reference

N/A — geen design voor deze taak.

---

### In scope

- Fallback-logica in `InternetaakDetailsService` die partij-adressen ophalt wanneer betrokkene geen eigen adressen heeft
- Graceful degradation bij fouten (API niet bereikbaar, partij niet gevonden)
- Logging van fallback-gebruik

### Out of scope

- Wijzigingen aan `OpenKlantApiClient` methode (Task 1)
- Frontend wijzigingen
- Herkomst-indicatie tonen (expliciet niet gewenst per Feature AC)

---

### Request

Intern — geen wijziging aan het externe API contract.

### Response

Het bestaande `Internetaak`-response met nested `Klantcontact.Expand.HadBetrokkenen[].Expand.DigitaleAdressen` bevat nu ook adressen afkomstig van de partij (transparant voor de client).

### Events / Side Effects

Geen — read-only verrijking.

---

## Technical Approach

- Voeg een `EnrichBetrokkenenWithPartijDigitaleAdressenAsync` methode toe aan `InternetaakDetailsService`
- Roep deze aan na het laden van het klantcontact (na `EnrichWithZaakAsync`)
- Voor elke betrokkene in `klantcontact.Expand.HadBetrokkenen`:
  - Als `betrokkene.Expand.DigitaleAdressen` null of leeg
  - EN `betrokkene.Expand.WasPartij?.Uuid` niet null
  - → Roep `GetPartijDigitaleAdressenAsync(wasPartij.Uuid)` aan
  - → Zet resultaat in `betrokkene.Expand.DigitaleAdressen`
- Wrap in try-catch voor graceful degradation

### Aggregates & Invariants

- Aggregates: Betrokkene, Partij (read-only)
- Invariants: Eigen adressen van betrokkene worden nooit overschreven; partij-lookup wordt alleen uitgevoerd bij afwezigheid eigen adressen

---

## Impacted Areas

- Backend: `InterneTaakAfhandeling.Web.Server/Features/InterneTaakDetails/InternetaakDetailsService.cs`
- Frontend: Geen
- Database: N/A
- APIs: Open Klant API (indirect via `GetPartijDigitaleAdressenAsync`)
- External systems: Open Klant

---

## Gherkin

```gherkin
Feature: Fallback digitale adressen via partij

  Scenario: Betrokkene zonder eigen adressen met gekoppelde partij die adressen heeft
    Given een betrokkene zonder eigen digitale adressen
    And de betrokkene is gekoppeld aan een partij met een e-mailadres en telefoonnummer
    When de medewerker het contactverzoek opent
    Then ziet de medewerker het e-mailadres en telefoonnummer van de partij in het behandelscherm

  Scenario: Betrokkene met eigen digitale adressen
    Given een betrokkene met een eigen e-mailadres
    And de betrokkene is gekoppeld aan een partij met een ander e-mailadres
    When de medewerker het contactverzoek opent
    Then ziet de medewerker alleen het eigen e-mailadres van de betrokkene

  Scenario: Betrokkene zonder eigen adressen en zonder partij-identifier
    Given een betrokkene zonder eigen digitale adressen
    And de betrokkene is niet gekoppeld aan een partij
    When de medewerker het contactverzoek opent
    Then blijft de contactsectie leeg

  Scenario: Betrokkene zonder eigen adressen met partij zonder adressen
    Given een betrokkene zonder eigen digitale adressen
    And de betrokkene is gekoppeld aan een partij zonder digitale adressen
    When de medewerker het contactverzoek opent
    Then blijft de contactsectie leeg

  Scenario: Partij-lookup faalt door onbereikbare API
    Given een betrokkene zonder eigen digitale adressen
    And de betrokkene is gekoppeld aan een partij
    And de Open Klant API is niet bereikbaar voor de partij-lookup
    When de medewerker het contactverzoek opent
    Then blijft de contactsectie leeg
    And wordt de fout gelogd zonder de medewerker te hinderen
```

---

## Edge Cases & Risks

- Open Klant API onbereikbaar bij fallback → graceful degradation (log warning, geen foutmelding naar gebruiker)
- Betrokkene heeft `WasPartij` met UUID maar partij bestaat niet → behandel als lege adressen
- Meerdere betrokkenen bij hetzelfde klantcontact → fallback per betrokkene afzonderlijk uitvoeren
- Performance: extra API-call alleen wanneer nodig (lazy fallback)

---

## Observability

- Logs: `LogInformation` wanneer fallback wordt uitgevoerd, `LogWarning` bij falen van partij-lookup
- Metrics: Geen nieuwe metrics
- Alerts: Geen

---

## Dependencies

Depends on Task 1 (#445): `GetPartijDigitaleAdressenAsync` methode moet beschikbaar zijn.

---

## Rollout

- Feature flag: N/A
- Backward compatibility: Volledig backward-compatible — bestaand gedrag ongewijzigd wanneer betrokkene eigen adressen heeft
- Data migration: N/A

---

## Definition of Done

- [ ] Contracts match implementation
- [ ] All Gherkin scenarios covered by tests (Phase 2 E2E)
- [ ] Edge cases handled
- [ ] Observability in place
- [ ] Code reviewed
- [ ] Linting passes (`npm run lint:ci`)
- [ ] Formatting passes (`npm run format:ci`)

---

## Test Mapping

| Gherkin Scenario | Test file | Status |
| ---------------- | --------- | ------ |
| Betrokkene zonder eigen adressen met gekoppelde partij die adressen heeft | Phase 2 E2E | Pending |
| Betrokkene met eigen digitale adressen | Phase 2 E2E | Pending |
| Betrokkene zonder eigen adressen en zonder partij-identifier | Phase 2 E2E | Pending |
| Betrokkene zonder eigen adressen met partij zonder adressen | Phase 2 E2E | Pending |
| Partij-lookup faalt door onbereikbare API | Phase 2 E2E | Pending |

### PR DESCRIPTION
**Pull request by nijmra**
## Summary

Wanneer een betrokkene bij een contactverzoek geen eigen digitale adressen heeft maar
wel gekoppeld is aan een partij, toont ITA nu de digitale adressen van die partij als
fallback. Dit zorgt ervoor dat medewerkers altijd contactgegevens zien wanneer die
beschikbaar zijn — ongeacht of ze bij de betrokkene of partij zijn opgeslagen.

Onderdeel van Feature #367.

## Changes

- **Open Klant API-client uitbreiding**: nieuwe `GetPartijDigitaleAdressenAsync` methode
  die `GET /partijen/{uuid}?expand=digitaleAdressen` aanroept en de response
  mapt naar het bestaande `DigitaleAdres`-model.
- **Server-side fallback-logica**: `InternetaakDetailsService` voert de partij-lookup
  alleen uit wanneer de betrokkene geen eigen adressen heeft én een `wasPartij.uuid`
  beschikbaar is. Bij fouten graceful degradation (log + lege lijst).
- **Unit tests**: twee nieuwe testprojecten (`Common.Test`, `Web.Server.Test`) met xUnit +
  Moq die de API-client en de fallback-logica volledig dekken.

## Test plan

- [x] Partij met digitale adressen → adressen worden teruggegeven
- [x] Partij zonder digitale adressen → lege lijst
- [x] Partij niet gevonden → fout gelogd, lege lijst
- [x] Betrokkene zonder eigen adressen + partij met adressen → partij-adressen getoond
- [x] Betrokkene met eigen adressen → alleen eigen adressen getoond
- [x] Betrokkene zonder adressen en zonder partij-identifier → contactsectie leeg
- [x] Betrokkene zonder adressen + partij zonder adressen → contactsectie leeg
- [x] Partij-lookup faalt → graceful degradation, fout gelogd
- [ ] E2E tests (Phase 2, post-deploy — Task #447)

## Related

Closes #445
Closes #446

### CODE REVIEW

**Review by nijmra — COMMENTED** [review]
## PR Review — #450: feat(open-klant): digitale adressen partij fallback

**Linked tasks:** #445, #446 | **Feature:** #367
**Spec coverage:** 8/8 Gherkin scenarios covered by unit tests

| Dimension | Result | Notes |
|-----------|--------|-------|
| Spec adherence | ✅ PASS | All 3 (#445) + 5 (#446) scenarios have matching unit tests |
| Contract compliance | ✅ PASS | `GET /partijen/{uuid}?expand=digitaleAdressen` per Task spec; `PartijExpand` model correct |
| Impacted Areas | ✅ PASS | Changes within declared scope |
| Code quality | ✅ PASS | Interface-driven, proper naming, nullable, structured logging |
| Test coverage | ✅ PASS | Happy path + error/edge paths for both client and service |
| DoD | ✅ PASS | Contracts match, edge cases handled, observability in place |

**Verdict: APPROVE WITH COMMENTS**

---

### Agree with Copilot findings

I agree with Copilot's inline comment on `OpenKlantApiClient.cs:508`: the generic `catch (Exception)` swallows `OperationCanceledException`. The service layer has `catch (OperationCanceledException) { throw; }` which is correct in intent — but unreachable in practice since the client returns `[]` instead of propagating. Add `catch (OperationCanceledException) { throw; }` in the client as well. Also: `using var response = await ...` for proper disposal.

The PR description endpoint mismatch (Copilot comment #2) is a documentation issue only — the implementation correctly follows the Task #445 contract.

### Additional suggestions (non-blocking)

1. **Observability:** Consider `LogDebug` instead of `LogInformation` for the routine "Fetching digitale adressen" message in `GetPartijDigitaleAdressenAsync` — it fires on every partij lookup and could be noisy. The fallback `LogInformation` in the service is appropriate (conditional).

2. **First unit tests in the solution 🎉** — This addresses the "No unit test coverage" pain point in TECH.md. `MockHttpMessageHandler` is a solid reusable foundation.

File: InterneTaakAfhandeling.Common/Services/OpenKlantApi/OpenKlantApiClient.cs [hunk 1]
```diff
@@ -481,6 +482,30 @@ public async Task<List<Partij>> GetPartijenByBsnAsync(string bsn)
...
+            var partij = await response.Content.ReadFromJsonAsync<Partij>();
+
+            return partij?.Expand?.DigitaleAdressen ?? [];
+        }
+        catch (Exception ex)
+        {
+            _logger.LogWarning(ex, "Error fetching digitale adressen for partij {PartijUuid}: {Message}", partijUuid, ex.Message);
+            return [];
+        }
+    }
```
[the remark this answers is not in the document]
Comment (nijmra) [reply InterneTaakAfhandeling.Common/Services/OpenKlantApi/OpenKlantApiClient.cs:512]: Fixed in 37abe16: added `catch (OperationCanceledException) { throw; }` and changed to `using var response` for proper disposal.

File: InterneTaakAfhandeling.Common/Services/OpenKlantApi/OpenKlantApiClient.cs [hunk 2]
```diff
@@ -481,6 +482,30 @@ public async Task<List<Partij>> GetPartijenByBsnAsync(string bsn)
...
 
+    public async Task<List<DigitaleAdres>> GetPartijDigitaleAdressenAsync(string partijUuid)
+    {
+        try
+        {
+            _logger.LogInformation("Fetching digitale adressen for partij {PartijUuid}", partijUuid);
+
+            var response = await _httpClient.GetAsync($"partijen/{partijUuid}?expand=digitaleAdressen");
+
+            if (!response.IsSuccessStatusCode)
```
[the remark this answers is not in the document]
Comment (nijmra) [reply InterneTaakAfhandeling.Common/Services/OpenKlantApi/OpenKlantApiClient.cs]: Fixed: PR description updated to match the implementation (`GET /partijen/{uuid}?expand=digitaleAdressen`). The Task #445 spec already specified the expand approach.

File: InterneTaakAfhandeling.Web.Server/Features/InterneTaakDetails/InternetaakDetailsService.cs [hunk 3]
```diff
@@ -86,5 +88,41 @@ private async Task EnrichWithZaakAsync(Internetaak? interntaak)
...
+                try
+                {
+                    _logger.LogInformation("Fallback: ophalen digitale adressen van partij {PartijUuid} voor betrokkene {BetrokkeneUuid}",
+                        partijUuid, betrokkene.Uuid);
+
+                    var partijAdressen = await _openKlantApiClient.GetPartijDigitaleAdressenAsync(partijUuid);
+
+                    betrokkene.Expand ??= new BetrokkeneExpand();
+                    betrokkene.Expand.DigitaleAdressen = partijAdressen;
+                }
```
[the remark this answers is not in the document]
Comment (nijmra) [reply InterneTaakAfhandeling.Web.Server/Features/InterneTaakDetails/InternetaakDetailsService.cs:121]: Fixed in 37abe16: added a `Dictionary<string, List<DigitaleAdres>> partijAdressenCache` scoped to the enrichment method. Only the first betrokkene for a given partijUuid triggers the API call; subsequent ones reuse the cached result.

**Review by mstokericatt — COMMENTED** [review]
I have some doubts about the tests. 
Since there is pressure to include this feature in the next release I made a separate pr with just the only the code changes and merged that. We'll discuss the tests later.
